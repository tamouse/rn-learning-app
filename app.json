{
  "expo": {
    "sdkVersion": "25.0.0"
  },
  "name": "rnlearningapp",
  "displayName": "RN APP"
}