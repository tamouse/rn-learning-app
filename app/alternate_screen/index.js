/* Alternate Screen, to demo StackNavigation */

import PropTypes from 'prop-types'
import React from 'react'
import { Button, Text, View } from 'react-native'
import styles from '../styles'

export default class AltScreen extends React.Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired // the navigation object from RootNavigator
  }

  goBack = () => this.props.navigation.goBack()
  gotoHomeScreen = () => this.props.navigation.navigate('Home')

  render() {
    return (
      <View style={styles.screen}>
        <Text style={styles.text}>
          Alternate Screen
        </Text>
        <Button
          title={`Go to Home Screen`}
          onPress={this.gotoHomeScreen} />
        <Button
          title={`Return`}
          onPress={this.goBack}/>
      </View>
    )
  }
}
