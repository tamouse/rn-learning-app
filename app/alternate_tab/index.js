import { StackNavigator } from "react-navigation";
import AltHomeScreen from "./home_screen";
import AltAltScreen from "./alt_screen";

export default (AltStack = StackNavigator(
  {
    AltHome: { screen: AltHomeScreen },
    AltAlt: { screen: AltAltScreen }
  },
  {
    navigationOptions: {
      title: "Alt"
    }
  }
));
