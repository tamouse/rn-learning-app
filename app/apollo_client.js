import { ApolloClient } from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import { InMemoryCache } from 'apollo-cache-inmemory'

const link = createHttpLink({
  uri: 'http://lvh.me:3000/shasta/graphql'
})

const cache = new InMemoryCache()

export const client = new ApolloClient({
  link,
  cache
})

export default client
