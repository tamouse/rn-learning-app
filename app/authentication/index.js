import LoginContainer from "./login_container"
export const Login = LoginContainer

export const isLoggedIn = () => {
  return false
}

export const userInfo = () => {
  return {}
}
