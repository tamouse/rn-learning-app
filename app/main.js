import React from "react"
import RootNavigator from "./routes"

export class MainApp extends React.Component {
  render() {
    return <RootNavigator />
  }
}

export default MainApp