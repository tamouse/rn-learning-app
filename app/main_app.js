import PropTypes from 'prop-types'
import React from 'react'
import { Text, View } from 'react-native'
import styles from './styles'
import { StackNavigator } from 'react-navigation'

class MainApp extends React.Component {
  static propTypes = {
    appName: PropTypes.string
  }

  render() {
    console.log(this.props)
    return (
      <View style={styles.screen}>
        <Text style={styles.defaultText}>
          This is the main app.
        </Text>
        <Text>
          It is called {this.props.appName}!
        </Text>
      </View>
    )
  }
}

export default StackNavigator({
  MainApp: {
    screen: MainApp
  }
})