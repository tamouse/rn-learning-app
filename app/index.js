import React from 'react'
import { isLoggedIn, userInfo, Login } from "./authentication"
import { MainApp } from './main'

export class App extends React.Component {

  render() {
    const props = this.props
    if (isLoggedIn()) {
      return (<MainApp {...props} userInfo={userInfo()} />)
    } else {
      return (<Login {...props}/>)
    }
  }
}

export default App
