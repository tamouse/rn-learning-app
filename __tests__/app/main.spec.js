import React from "react"
import renderer from "react-test-renderer"

import { MainApp } from "../../app/main"

describe("Testing MainApp", ()=>{
  it('matches the snapshot', function () {
    expect(renderer.create(
      <MainApp/>
    )).toMatchSnapshot()
  });
})