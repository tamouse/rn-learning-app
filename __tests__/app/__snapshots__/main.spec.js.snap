// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainApp matches the snapshot 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    collapsable={undefined}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID={undefined}
    >
      <View
        collapsable={false}
        removeClippedSubviews={false}
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={undefined}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#EFEFF4",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "blue",
                          "fontSize": 30,
                        }
                      }
                    >
                      Main Tab
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={undefined}
                    >
                      The Main Home Screen
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    >
                      to Account 
                      Unknown!
                    </Text>
                    <View
                      accessibilityComponentType="button"
                      accessibilityLabel={undefined}
                      accessibilityTraits={
                        Array [
                          "button",
                        ]
                      }
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View
                        style={
                          Array [
                            Object {},
                          ]
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          disabled={undefined}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "padding": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          goto second screen
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType="button"
                      accessibilityLabel={undefined}
                      accessibilityTraits={
                        Array [
                          "button",
                        ]
                      }
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View
                        style={
                          Array [
                            Object {},
                          ]
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          disabled={undefined}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "padding": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          go to modal view
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType="button"
                      accessibilityLabel={undefined}
                      accessibilityTraits={
                        Array [
                          "button",
                        ]
                      }
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View
                        style={
                          Array [
                            Object {},
                          ]
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          disabled={undefined}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "padding": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Visit Generic
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                collapsable={undefined}
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "backgroundColor": "#F7F7F7",
                    "borderBottomColor": "#A7A7AA",
                    "borderBottomWidth": 0.5,
                    "height": 64,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 20,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "flexDirection": "row",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      collapsable={undefined}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "bottom": 0,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "left": 70,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 70,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                              "flexDirection": "row",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "color": "black",
                                "fontSize": 20,
                                "fontWeight": "900",
                              }
                            }
                          >
                            main
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      collapsable={undefined}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "bottom": 0,
                          "flexDirection": "row",
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "flex": 1,
                            "justifyContent": "center",
                            "padding": 20,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "black",
                                "fontSize": 15,
                              },
                              undefined,
                              Object {
                                "fontFamily": "FontAwesome",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                      </View>
                    </View>
                    <View
                      collapsable={undefined}
                      pointerEvents="box-none"
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "bottom": 0,
                          "flexDirection": "row",
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "flex": 1,
                            "justifyContent": "center",
                            "padding": 20,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "white",
                            }
                          }
                        >
                          empty
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={null}
      testID={undefined}
    />
    <View
      style={null}
      testID={undefined}
    />
  </View>
  <View
    collapsable={undefined}
    style={undefined}
  >
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "zIndex": 999,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "tomato",
                  }
                }
              >
                Home
                !
              </Text>
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Home
                !
              </Text>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(255, 99, 71, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            main
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "tomato",
                  }
                }
              >
                Alt
                !
              </Text>
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Alt
                !
              </Text>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(128, 128, 128, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            Alt
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "tomato",
                  }
                }
              >
                Logout
                !
              </Text>
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Logout
                !
              </Text>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(128, 128, 128, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
